test/validator/gitignore_test.dart

```diff
@@ -121,6 +128,17 @@ void main() {
       exit_codes.DATA,
       workingDirectory: packageRoot,
     );
+
+    // Check that you can unignore certain dirs from a higher level gitignore.
+    await d.dir('reporoot', [
+      d.file('.gitignore', '*.txt\n!myapp/'),
```

Comment:
I don't think this does whatever you think it does.

`myapp/` is not ignored here, so it cannot meaningfully be negated/unignored.

We could test that `"/*\n!myapp/"` leads to no warnings. That is already the case.
```suggestion
      d.file('.gitignore', '/*\n!myapp/'),
```